### ISSUE TITLE
Enh Req: BM Counts in VM Names

### ISSUE THREAD

**Issue by timothyeharris** [GoogleCloudPlatform/PerfKitBenchmarker#141]
When running a set of benchmarks, it's difficult to know (looking at the console screen) how many benchmarks have run and how many are left.  It would be kind of nice if the VM names associated with the benchmarks could communicate progress to you.

For example, today they're all named perfkit-xxxxxxxx-x.  Maybe they could be changed to pkBB-TT-xxxxxxxx-x where BB represents the current number of benchmark(s) that have/are executed and TT represents the total number of benchmarks to be executed.  This would still leave 8 characters for the random runID and would not change the overall string length.

(I recognize this is probably a super low priority enhancement request)

**Comment by voellm** [comment]
Perhaps we should prepend all the messages with something like
<timecode> [RUN 1/22[ ...
.
.
.
<timecode> [RUN 10/22] ...

On Tue, Mar 3, 2015 at 3:36 PM, Tim Harris <notifications@github.com> wrote:

> When running a set of benchmarks, it's difficult to know (looking at the
> console screen) how many benchmarks have run and how many are left. It
> would be kind of nice if the VM names associated with the benchmarks could
> communicate progress to you.
>
> For example, today they're all named perfkit-xxxxxxxx-x. Maybe they could
> be changed to pkBB-TT-xxxxxxxx-x where BB represents the current number of
> benchmark(s) that have/are executed and TT represents the total number of
> benchmarks to be executed. This would still leave 8 characters for the
> random runID and would not change the overall string length.
>
> (I recognize this is probably a super low priority enhancement request)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues/141>.
>



-- 
Anthony F. Voellm (aka Tony)
Google Voice:  (650) 516-7382
<https://www.google.com/voice/b/0?pli=1#phones>
Blog: http://perfguy.blogspot.com/

**Comment by timothyeharris** [comment]
That would work too. If it were colored, even better.

**Comment by timothyeharris** [comment]
The only other argument for the VM name approach is the ability to check progress via the portal if you walk away from the console/terminal session.

**Comment by timothyeharris** [comment]
If you're going to prepend the output messages, instead of RUN XX/XX, it would be great if each line could be RUN_ID XX/XX.  Sometimes finding the runID is a PITA :-)

**Comment by skschneider** [comment]
Pull #157 modified the log format. For executing benchmarks, it is now "<time> <run_uri> <thread_name> <benchmark>(<sequence>/<total>) <level> <message>", and the verbose log file still has the file:line prior to the level. Note that <sequence> is a static number assigned to the benchmark when a thread is selected to run it; it is not updated at each log message to reflect the number of completed benchmarks in parallel runs.

**Comment by voellm** [comment]
This is in the dev branch now and should make it to master is the next release.
